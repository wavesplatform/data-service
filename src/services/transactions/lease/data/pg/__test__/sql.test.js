const sql = require('../sql');
const filters = {
  sender: 'sender',
  recipient: 'recipient',
  timeStart: 'timeStart',
  timeEnd: 'timeEnd',
  sort: 'sort',
  limit: 1,
};
describe('Sql builder', () => {
  describe(' many', () => {
    it('covers case with all filters (without after)', () => {
      expect(sql.many(filters)).toMatchSnapshot();
    });
    it('covers case with all filters with after', () => {
      expect(
        sql.many({
          ...filters,
          after: {
            timestamp: 'timestamp',
            id: 'id',
            sortDirection: 'sortDirection',
          },
        })
      ).toMatchSnapshot();
    });
  });
  describe(' one', () => {
    it('works', () => {
      expect(sql.one('id')).toMatchSnapshot();
    });
  });
});
