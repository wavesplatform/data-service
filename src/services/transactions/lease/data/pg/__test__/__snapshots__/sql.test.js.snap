// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql builder  many covers case with all filters (without after) 1`] = `"select \\"t\\".\\"height\\" as \\"height\\", \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\", \\"t\\".\\"time_stamp\\" as \\"time_stamp\\", \\"t\\".\\"signature\\" as \\"signature\\", \\"t\\".\\"proofs\\" as \\"proofs\\", \\"t\\".\\"tx_version\\" as \\"tx_version\\", t.fee * 10^(-8) as \\"fee\\", \\"t\\".\\"sender\\" as \\"sender\\", \\"t\\".\\"sender_public_key\\" as \\"sender_public_key\\", t.amount * 10^(-8) as \\"amount\\", \\"t\\".\\"recipient\\" as \\"recipient\\" from \\"txs_8\\" as \\"t\\" where \\"sender\\" = 'sender' and \\"recipient\\" = 'recipient' and \\"t\\".\\"time_stamp\\" >= 'timeStart' and \\"t\\".\\"time_stamp\\" <= 'timeEnd' order by \\"t\\".\\"time_stamp\\" asc, \\"t\\".\\"id\\" asc limit 1"`;

exports[`Sql builder  many covers case with all filters with after 1`] = `"select \\"t\\".\\"height\\" as \\"height\\", \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\", \\"t\\".\\"time_stamp\\" as \\"time_stamp\\", \\"t\\".\\"signature\\" as \\"signature\\", \\"t\\".\\"proofs\\" as \\"proofs\\", \\"t\\".\\"tx_version\\" as \\"tx_version\\", t.fee * 10^(-8) as \\"fee\\", \\"t\\".\\"sender\\" as \\"sender\\", \\"t\\".\\"sender_public_key\\" as \\"sender_public_key\\", t.amount * 10^(-8) as \\"amount\\", \\"t\\".\\"recipient\\" as \\"recipient\\" from \\"txs_8\\" as \\"t\\" where \\"sender\\" = 'sender' and \\"recipient\\" = 'recipient' and (t.time_stamp, t.id) > ('timestamp', 'id') and \\"t\\".\\"time_stamp\\" >= 'timeStart' and \\"t\\".\\"time_stamp\\" <= 'timeEnd' order by \\"t\\".\\"time_stamp\\" asc, \\"t\\".\\"id\\" asc limit 1"`;

exports[`Sql builder  one works 1`] = `"select \\"t\\".\\"height\\" as \\"height\\", \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\", \\"t\\".\\"time_stamp\\" as \\"time_stamp\\", \\"t\\".\\"signature\\" as \\"signature\\", \\"t\\".\\"proofs\\" as \\"proofs\\", \\"t\\".\\"tx_version\\" as \\"tx_version\\", t.fee * 10^(-8) as \\"fee\\", \\"t\\".\\"sender\\" as \\"sender\\", \\"t\\".\\"sender_public_key\\" as \\"sender_public_key\\", t.amount * 10^(-8) as \\"amount\\", \\"t\\".\\"recipient\\" as \\"recipient\\" from \\"txs_8\\" as \\"t\\" where \\"id\\" = 'id'"`;
